lib/models/addon.js

```diff
@@ -165,6 +165,86 @@ function warn(message) {
   @param {Project} project The current project (deprecated)
 */
 let addonProto = {
+  /**
+    The name of this addon.
+
+    @public
+    @final
+    @type String
+    @property name
+  */
+
+  /**
+    The root directory where this addon is located.
+
+    @public
+    @final
+    @type String
+    @property root
+  */
+
+  /**
+    The host app instance.
+
+    **Note**: this property will only be present on addons that are a direct dependency
+    of the application itself, not of other addons. It is also only available after the
+    base `included()` hook has been run.
+
+    @public
+    @final
+    @type EmberApp
+    @property app
+  */
+
+  /**
+    The root {{#crossLink "Project"}}project{{/crossLink}} to which this addon belongs.
+
+    @public
+    @final
+    @type Project
+    @property project
+  */
+
+  /**
+    This addon's parent.
+
+    If the addon is a direct dependency of an application, then `parent` will be the
+    corresponding {{#crossLink "Project"}}project{{/crossLink}} instance. If it's a
+    dependency of another addon, then `parent` will be a reference to that addon.
+
+    @public
+    @final
+    @type Project|Addon
+    @property parent
+  */
+
+  /**
+    The set of addons that this addon itself depends on.
```

Comment:
Might want to mention that the list is built up from `dependencies` and `ember-addon.paths` items in the `package.json`? What do you think?

Reply:
👍 Yep, that makes sense to me